### PR TITLE
Allow AIs to check for valid enemy startpos

### PR DESCRIPTION
**Pull request by mtroyka**
Since startboxes are no longer exposed to AIs through rules params they are no longer able to infer enemy start positions. To get around this I added an "ai_is_valid_enemy_startpos:" ai callin so that they can  do the same thing via rules calls.

Note: I haven't tested this because there's no skirmish mode.. additionally it's possible that Spring.GetAllyTeamList() and Spring.GetTeamList() do not return an ordinal list as documented but instead use the ally/team ids as keys, which would require changing the code.

### CODE REVIEW

**Comment by Licho1** [comment]
Hmm so we should add boxes to startscript again?

**Comment by mtroyka** [comment]
Boxes previously cluttered up gamerulesparams like nobody's business, so I'd have to lean toward "no". It's also more complicated to have every ai parse the boxes instead of just using rules calls and letting lua do it.

For reference as far as I know only ZKGBAI parsed the startboxes directly anyway, circuit just uses the ai_is_valid_startpos call, but circuit doesn't check enemy positions either. KGB only tests metal spots so the number of tests generally isn't huge.

**Comment by Anarchid** [comment]
Another option would be to add a CallRules for AI's to query boxes without cluttering up GRP.

File: LuaRules/Gadgets/start_boxes.lua [hunk 1]
```diff
@@ -293,10 +300,33 @@ function gadget:RecvSkirmishAIMessage(teamID, dataStr)
...
-		return "0"
+		-- for checking enemy startpos
+		local enemyboxes = {}
+		local allyteamid = Spring.GetAllyTeamID(teamID)
+		local allyteams = Spring.GetAllyTeamList()
+		
+		for _,value in pairs(allyteams) do
+			if value ~= allyteamid then
+				local enemyteams = Spring.GetTeamList(value)
+				enemyboxes[Spring.GetTeamRulesParam(enemyteams[1], "start_box_id")] = true
```
Comment (sprunk) [inline LuaRules/Gadgets/start_boxes.lua]: This reveals the true box after shuffling.

**Pull request merged**